vllm/platforms/tpu.py

```diff
@@ -149,3 +149,5 @@ def validate_request(
                       SamplingParams) and params.guided_decoding is not None:
             raise ValueError("Structured output is not supported on "
                              f"{cls.device_name}.")
+        if params.sampling_type == SamplingType.RANDOM_SEED:
+            raise ValueError("Torch XLA does not support per-request seed.")
```

Comment:
You still need to check that params is SamplingParams, maybe you could wrap both cases in `if isinstance(params, SamplingParams)` first